### ISSUE TITLE
Breaking change (`0.21.1` -> `0.22.0`): Unable to reference sibling task files

### ISSUE THREAD

**Issue by m-roberts** [nat-n/poethepoet#165]
`pyproject.toml`
```toml
[tool.poe]
include = [
  "tasks/first.toml",
  "tasks/second.toml",
]
```

`tasks/first.toml`
```toml
[tool.poe.tasks.parent-task]
help = "Run parent task"
sequence = [
  { cmd = "true" },
  "child-task",
]
```

`tasks/second.toml`
```toml
[tool.poe.tasks.child-task]
help = "Run child task"
cmd = "true"
```

```shell
$ poe parent-task
```
will now error where it didn't before, as `first.toml` cannot see `second.toml`, despite both being included in `pyproject.toml`.

**Comment by nat-n** [comment]
Hi @m-roberts, thanks for reporting this. I hadn't considered that use case, so didn't have a test for it, but it makes sense.

It looks like my recent refactor of config loading logic changed the order of validation logic with this unintended consequence. But it should be easy to fix :)

**Comment by nat-n** [comment]
@m-roberts Shipped a fix in 0.22.1 🚀
